src/time-picker/time-picker.jsx

```diff
@@ -103,51 +110,83 @@ const TimePicker = React.createClass({
 
   getInitialState() {
     return {
-      time: this.props.defaultTime || emptyTime,
+      time: this._isControlled() ? this._getControlledTime() : this.props.defaultTime,
       dialogTime: new Date(),
       muiTheme: this.context.muiTheme || getMuiTheme(),
     };
   },
 
+  componentWillReceiveProps(nextProps, nextContext) {
+    const newState = this.state;
+    if (nextContext.muiTheme) {
+      newState.muiTheme = nextContext.muiTheme;
+    }
+    if (this._isControlled()) {
+      let newTime = this._getControlledTime(nextProps);
+      if (!DateTime.isEqualTime(newTime, this.state.time)) {
+        newState.time = newTime;
+      }
+    }
+    this.setState(newState);
+  },
+
   windowListeners: {
     'keyup': '_handleWindowKeyUp',
   },
 
-  formatTime(date) {
-    let hours = date.getHours();
-    let mins = date.getMinutes().toString();
-
-    if (this.props.format === 'ampm') {
-      let isAM = hours < 12;
-      hours = hours % 12;
-      let additional = isAM ? ' am' : ' pm';
-      hours = (hours || 12).toString();
-
-      if (mins.length < 2 ) mins = '0' + mins;
-
-      if (this.props.pedantic) {
-        // Treat midday/midnight specially http://www.nist.gov/pml/div688/times.cfm
-        if (hours === '12' && mins === '00') {
-          return additional === ' pm' ? '12 noon' : '12 midnight';
+  formatTime(date, format = 'ampm', pedantic = false) {
```

Comment:
That seems to be the same code as in ` src/utils/date-time.js`.
Can we remove this method for this component and use the on of `src/utils/date-time.js`?